src/hotspot/share/prims/jvmtiRedefineClasses.cpp

```diff
@@ -1823,6 +1823,11 @@ jvmtiError VM_RedefineClasses::merge_cp_and_rewrite(
     return JVMTI_ERROR_INTERNAL;
   }
 
+  // ensure merged constant pool size does not overflow u2
+  if (merge_cp_length > 0xFFFF) {
+    return JVMTI_ERROR_INTERNAL;
```

Comment:
Doesn't it make a sense to add some logging there so user could easier understand the problem?

Reply:
Seems like a good idea, you can use log_warning(redefine, class, constantpool)("...")

Reply:
Done.